pkg/app/app_reconcile.go

```diff
@@ -267,3 +270,16 @@ func (a *App) setDeleteCompleted(result exec.CmdRunResult) {
 func (a *App) removeAllConditions() {
 	a.app.Status.Conditions = nil
 }
+
+func (a *App) setUsefulErrorMessage(result exec.CmdRunResult) {
+	switch {
```

Comment:
can this just have two cases? 

case Stderr != "" => use stderr
case default => ErrorStr()

Seems like the only difference between FriendlyDescription and ErrorStr is the preamble about what failed. If that is important info, maybe then we duplicate make the default case above use friendly desc instead of ErrorStr().

Reply:
I ran into this issue previously of not being able to populate UsefulErrorMessage from ErrorStr() when trying to create a deletion error. I cannot reproduce it anymore so wondering if I had something deployed that was inconsistent with what I had in source at time, leading me to add the third case.

But yeah, makes sense. Thanks for pointing that out.